test_project/test_app/tests/test_utils.py

```diff
@@ -0,0 +1,205 @@
+from django.test import TestCase
+
+from autocompleter import utils
+
+
+class TestFacetHash(TestCase):
+    def test_facet_identity_hash(self):
+        """
+        Hashing a facet should equal the hash of an earlier call
+        """
+        facets = [
+            {
+                'type': 'or',
+                'facets': [{'key': 'sector', 'value': 'Technology'}]
+            }
+        ]
+        first_hash = utils.hash_facets(facets)
+        second_hash = utils.hash_facets(facets)
+        self.assertEqual(first_hash, second_hash)
+
+    def test_hashing_facet_type(self):
+        """
+        Facet list with same sub facets but different type should not have equal hashes
+        """
+        and_facet = [
+            {
+                'type': 'and',
+                'facets': [{'key': 'sector', 'value': 'Technology'}]
+            }
+        ]
+
+        or_facet = [
+            {
+                'type': 'or',
+                'facets': [{'key': 'sector', 'value': 'Technology'}]
+            }
+        ]
+        and_hash = utils.hash_facets(and_facet)
+        or_hash = utils.hash_facets(or_facet)
+        self.assertNotEqual(and_hash, or_hash)
+
+    def test_hashing_order(self):
+        """
+        Facets with identical key/values in different order should still have same hash
+        """
+        facet_1 = [
```

Comment:
watch the length of all the new data structures you added to this file ... keep everything < 100 chars